1.18.2: cmake builds fail

#527 has returned between 1.18.0–1.18.2.

```
[    8s] CMake Error at cmake/test-functions.cmake:186 (add_executable):
[    8s]   Cannot find source file:
[    8s] 
[    8s]     poc_gh_517.cpp
[    8s] 
[    8s]   Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm
[    8s]   .ccm .cxxm .c++m .h .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90
[    8s]   .f95 .f03 .hip .ispc
```

--- a/Makefile.am
+++ b/Makefile.am
@@ -36,6 +36,8 @@ EXTRA_DIST = \
 	gtest/test_template.cpp \
 	gtest/test_upnpapi.cpp \
 	gtest/TODO \
+	scripts/comp-autotools.sh \
+	scripts/comp-cmake.sh \
 	scripts/find_out_bz2_name.sh \
 	scripts/home_keeping.sh \
 	scripts/parse_release.sh \

--- a/configure.ac
+++ b/configure.ac
@@ -581,6 +581,7 @@ RT_BOOL_ARG_ENABLE([samples], [yes], [compilation of upnp/sample/ code])
 # Checks for programs
 #
 AC_PROG_CC
+AC_PROG_CXX
 AM_PROG_CC_C_O
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR])
 LT_INIT

--- a/ixml/Makefile.am
+++ b/ixml/Makefile.am
@@ -53,15 +53,19 @@ upnpinclude_HEADERS	= \
 
 check_PROGRAMS          = \
 			test_document \
-			poc_gh_506
+			poc_gh_506 \
+			poc_gh_517
 
 TESTS                   = \
 			test/test_document.sh \
-			poc_gh_506
+			poc_gh_506 \
+			poc_gh_517
 
 test_document_SOURCES 	= test/test_document.c
 
 poc_gh_506_SOURCES      = test/poc_gh_506.c
+
+poc_gh_517_SOURCES      = test/poc_gh_517.cpp
 
 EXTRA_DIST		= test/test_document.sh test/testdata
 

